src/lib/execPromise.ts

```diff
@@ -37,33 +41,40 @@ export type ExecPromiseArg = {
   printOutput?: boolean | undefined;
 };
 
-export const execPromise = ({
+export const execPromise = async ({
   command,
   args,
   flags,
   options,
   spinnerMessage,
   timeout,
-  printOutput = false,
+  printOutput: printOutputArg = false,
 }: ExecPromiseArg): Promise<string> => {
+  const printOutput = isInteractive ? printOutputArg : false;
   const flagArgs = flags === undefined ? [] : flagsToArgs(flags);
   const allArgs = [...(args ?? []), ...flagArgs];
   const fullCommand = [command, ...allArgs].join(" ");
 
-  const commandToDisplay = IS_DEVELOPMENT
-    ? fullCommand
-    : fullCommand.replace(
-        /([\S\s]*--sk ).*([\S\s]*)/,
-        "$1<SECRET_KEY_IS_HIDDEN>$2"
-      );
+  const getCommandFailedMessage = (code: number | null = null) => {
+    return `Command: ${color.yellow(
+      IS_DEVELOPMENT
+        ? fullCommand
+        : fullCommand.replace(
+            /([\S\s]*--sk ).*([\S\s]*)/,
+            "$1<SECRET_KEY_IS_HIDDEN>$2"
+          )
+    )} ${code === null ? "failed" : `exited with code ${code}`}`;
```

Comment:
how could `code` be null?

Reply:
typescript api says it can be null. I just handle this possibility. Don't know how to actually achieve this